File: connectors/connector-github/src/main/java/com/taskadapter/connector/github/GithubTaskSaver.java
package com.taskadapter.connector.github;

import com.taskadapter.connector.common.BasicIssueSaveAPI;
import com.taskadapter.connector.definition.TaskId;
import com.taskadapter.connector.definition.exceptions.ConnectorException;
import org.eclipse.egit.github.core.Issue;
import org.eclipse.egit.github.core.service.IssueService;

import java.io.IOException;

public class GithubTaskSaver implements BasicIssueSaveAPI<Issue> {
    private final IssueService issueService;
    private final String userName;
    private final String projectKey;

    public GithubTaskSaver(IssueService issueService, String userName, String projectKey) {
        this.issueService = issueService;
        this.userName = userName;
        this.projectKey = projectKey;
    }

    public TaskId createTask(Issue issue) throws ConnectorException {
        var repositoryName = projectKey;
        try {
            var createdIssue = issueService.createIssue(userName, repositoryName, issue);
            return new TaskId((long) createdIssue.getNumber(), createdIssue.getNumber() + "");
        } catch (IOException e) {
            throw GithubUtils.convertException(e);
        }
    }

    public void updateTask(Issue issue) throws ConnectorException {
        try {
            issueService.editIssue(userName, projectKey, issue);
        } catch (IOException e) {
            throw GithubUtils.convertException(e);
        }
    }
}

File: connectors/connector-github/src/main/scala/com/taskadapter/connector/github/GithubTaskSaver.scala
package com.taskadapter.connector.github

import java.io.IOException

import com.taskadapter.connector.common.BasicIssueSaveAPI
import com.taskadapter.connector.definition.TaskId
import com.taskadapter.connector.definition.exceptions.ConnectorException
import org.eclipse.egit.github.core.Issue
import org.eclipse.egit.github.core.service.IssueService

final class GithubTaskSaver(var issueService: IssueService, val userName: String, val projectKey: String) extends BasicIssueSaveAPI[Issue] {
  @throws[ConnectorException]
  override def createTask(issue: Issue): TaskId = {
    val repositoryName = projectKey
    try {
      val createdIssue = issueService.createIssue(userName, repositoryName, issue)
      TaskId(createdIssue.getNumber, createdIssue.getNumber + "")
    } catch {
      case e: IOException =>
        throw GithubUtils.convertException(e)
    }
  }

  @throws[ConnectorException]
  override def updateTask(issue: Issue): Unit = try issueService.editIssue(userName, projectKey, issue)
  catch {
    case e: IOException =>
      throw GithubUtils.convertException(e)
  }
}